lib/view_model/active_record/nested_controller_base.rb

```diff
@@ -49,6 +49,50 @@ def write_association(serialize_context: new_serialize_context, deserialize_cont
     association_view
   end
 
+  def write_association_bulk(serialize_context: new_serialize_context, deserialize_context: new_deserialize_context, lock_owner: nil)
+    association_view = nil
+    pre_rendered = owner_viewmodel.transaction do
+      updates_by_parent_id, references = parse_bulk_update
+
+      updated_by_parent_viewmodel =
+        owner_viewmodel.replace_associated_bulk(
+          association_name,
+          updates_by_parent_id,
+          references:          references,
+          deserialize_context: deserialize_context,
+        )
+
+      views = updated_by_parent_viewmodel.flat_map { |_parent_viewmodel, updated_views| updated_views }
```

Comment:
This should be `Array.wrap` to avoid a type error, though empirically `flat_map` seems to handle this already.

```
irb(main):002:0> [1, [2], 3].flat_map { |x| x }
=> [1, 2, 3]
```

The return structure should be correct here. At least the intent is to preserve the singleton or collection nature through all the layers. So `replace_associated_bulk` should return `{ parent => child }` for the singleton case, and `{ parent => [children] }` for the collection case.

Reply:
Ah, my impression was that it hadn’t achieved that, I must have been looking in the wrong place